### PR TITLE
Add Token Extensions Support on Governance Program

### PR DESCRIPTION
**Pull request by crypt0miester**
This pull request was created by https://app.gib.work/i/Qd3CgojL/feature-add-token-extensions-support-on-governance-program in an attempt to solve a bounty. Payment for the bounty is immediately sent to the contributor after merge.

The following fixes are for the addressed issue in #5 

* building the program works `cargo build-sbf` on v1.18.12
* added tests to include the changes required and which was in the scope of work 
* previous tests work as intended. some minimal changes had to be done (in the previous tests) for token extensions to tests to succeed.

Please do not hesitate to ask me for any questions.

Regards, 
miester.

### CODE REVIEW

File: governance/program/src/state/realm.rs [hunk 1]
```diff
@@ -160,8 +160,11 @@ pub struct RealmV2 {
     /// Configuration of the Realm
     pub config: RealmConfig,
 
+    /// Check if Governing Token is spl_token_22
+    pub is_token_2022: bool,
```
Comment (asktree) [inline governance/program/src/state/realm.rs:164]: Is this really necessary? On the client you can just check the mint to see if its token22, and in the program you can just check if spl_token or spl_token_2022 program is passed in.

**Comment by asktree** [comment]
As far as I can tell the is_token_2022 field they added is not necessary and adding fields to the accounts is a real PITA. 

Aside from that everything made sense to me. I did not do an exhaustive check to make sure you handled every case that needs to be considered, I just looked at the changes that were made and they made sense to me.

File: governance/program/src/state/realm.rs (hunk 1 above)
Comment (crypt0miester) [reply governance/program/src/state/realm.rs:164]: makes sense. I always prefer to have a fast config read of downstream data. it is much easier to differentiate. but as you said, you can just check in the client. I will remove the is_token_2022 in the structs and the tests.

**Comment by crypt0miester** [comment]
> As far as I can tell the is_token_2022 field added is not necessary and adding fields to the accounts is a real PITA.
> 
> Aside from that everything made sense to me. I did not do an exhaustive check to make sure you handled every case that needs to be considered, I just looked at the changes that were made and they made sense to me.

thank you sir. I will do the necessary changes on the struct leave them as they were before.

**Comment by crypt0miester** [comment]
is_token_2022 in the struct fields has been removed.

**Comment by crypt0miester** [comment]
As promised, token_fees and transfer_hook now works with deposit and withdrawals.

previous tests were also passing.

**Pull request merged**